tests/Doctrine/Tests/DBAL/Schema/IndexTest.php

```diff
@@ -188,4 +189,40 @@ public function testOptions() : void
         self::assertSame('name IS NULL', $idx2->getOption('WHERE'));
         self::assertSame(['where' => 'name IS NULL'], $idx2->getOptions());
     }
+
+    /**
+     * @param int[] $lengths1
+     * @param int[] $lengths2
+     *
+     * @dataProvider spatialIndexProvider
+     */
+    public function testFulfilledSpatialIndex(?string $flag1, ?string $flag2, array $lengths1, array $lengths2, bool $expected) : void
+    {
+        $index1 = new Index('index1', ['geo_location'], false, false, [], ['lengths' => $lengths1]);
+        if ($flag1 !== null) {
+            $index1->addFlag($flag1);
+        }
+
+        $index2 = new Index('index2', ['geo_location'], false, false, [], ['lengths' => $lengths2]);
+        if ($flag2 !== null) {
+            $index2->addFlag($flag2);
+        }
+
+        self::assertSame($expected, $index1->isFullfilledBy($index2));
+        self::assertSame($expected, $index2->isFullfilledBy($index1));
+    }
+
+    public static function spatialIndexProvider() : Generator
```

Comment:
Please use `iterable` instead since PHPUnit doesn't need a more specific type, and it's just an implementation detail of this given provider.

Reply:
PHPUnit does not restrict the data provider return type. Do we have contribution guidelines doc where it describes what types must be used for data providers?

Reply:
No, but we have existing tests that could be used as an example.